lib/ansible/utils/collection_loader/_collection_finder.py

```diff
@@ -211,19 +225,26 @@ def find_module(self, fullname, path=None):
         try:
```

Comment:
Can this be refactored a bit? Having a big try-block is an antipattern. The loader factory selection will never cause an import error so it could be moved out and then the only thing that may cause this exception would be in try-except.
```python
if ...:
    initialize_loader = ...
elif ...:
    initialize_loader = ...
else:
    initialize_loader = _AnsibleCollectionLoader

try:
    loader = initialize_loader(fullname=fullname, path_list=path)
except ImportError:
    return None
```